### PR TITLE
Add support tag checker

### LINKED ISSUE #29: Feature request: tagged with semantic versioning but not released , tag detection support
I propose support tag mode.

Now some softwares are released by semver, but not all.
And other somes not released software but that is tagged with semver.

Therefore I request to support above .

### PR DESCRIPTION
**Pull request by tsuyoshicho**
Close #29.

- Add tag param (default off and false)
- Use git ls-remote --tags
- Treat refs list
- Add test case/test data and test PR create
- Update README

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -27,13 +27,27 @@ list_releases() {
...
-  head -n1 \
-)"
+
+LATEST_VERSION="${CURRENT_VERSION}"
+if [ "${INPUT_TAG:-false}" = "true" ]; then
+  echo "Check remote tag instead of release" >&2
+  LATEST_VERSION="$(\
+    git ls-remote -q --tags "https://github.com/${REPO}.git" | \
+    awk '{ print $2 }' | \
+    grep -oP '\d+\.\d+(\.\d+)*(-[^'\''\"\s]*)?$' | \
```
Comment (haya14busa) [inline entrypoint.sh:37]: Can you create a function which lists tags/releases name and pass the results to grep/sort/head instead of copying grep/sort/head comand?

**Review by haya14busa — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 1 above)
Comment (tsuyoshicho) [reply entrypoint.sh:37]: An... Okay fix it.

File: entrypoint.sh [hunk 2]
```diff
@@ -27,13 +27,23 @@ list_releases() {
...
+    list_releases |  jq -r '.[] | .tag_name'
+  fi
+}
+
 LATEST_VERSION="$(\
-  list_releases | \
-  jq -r '.[] | .tag_name' | \
-  grep -oP '\d+\.\d+(\.\d+)?(-[^'\''\"\s]*)?$' | \
+  list_versions | \
+  grep -oP '\d+\.\d+(\.\d+)*(-[^'\''\"\s]*)?$' | \
```
Comment (haya14busa) [inline entrypoint.sh:42]: oh, wait. I noticed that you changed the regex too. 
Can you revert the change or explain why you updated the regex?

**Review by haya14busa — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 1 above)
Comment (haya14busa) [reply entrypoint.sh]: Thanks now it's clearer :)

File: entrypoint.sh (hunk 2 above)
Comment (tsuyoshicho) [reply entrypoint.sh:42]: ## Why

Some tags have been given a small version that deviates from the rules of semver, so I changed it.

## Validity of change

Because, previouse commit split release(semver valid) and tag (semver deviate).

First, I think non-semver ruling tag(x.y.z.?) ignoring rule...but test case support non-semver(for -hoge suffix).
And I tried to support it.

Comment (haya14busa) [reply entrypoint.sh:42]: Hmm... I guess I don't get it. Can you list actual real-world examples who uses `x.y.z.?`?

Also, this repo doesn't intend to support non-semver in the first place, so can you propose a change separately?

Comment (haya14busa) [reply entrypoint.sh:42]: Note that `-xyz suffix` is actually valid semver. https://semver.org/#spec-item-10

Comment (tsuyoshicho) [reply entrypoint.sh:42]: Right, example

https://github.com/thinca/vim-themis/releases/tag/v1.5.2.1

and get

```
5.2.1
```

Comment (haya14busa) [inline entrypoint.sh:42]:
```suggestion
  grep -oP '\d+(\.\d+)+(-[^'\''\"\s]*)?$'| \
```

Comment (haya14busa) [reply entrypoint.sh:42]: Oh, I get it. Hmm... 1.5.2.1 is a super weird tag name...

Ideally, I'd reject such inputs, but maybe it's fine to accept. I left a suggestion to make the regex simpler but LGTM other than that.

Comment (tsuyoshicho) [reply entrypoint.sh:42]: Agree.
x.y.z.a ... can be simply drop away.

**Review by haya14busa — APPROVED** [review]

**Comment by haya14busa** [comment]
Thanks :+1:
